### PR TITLE
fix: re-add 'spawn' method for the multiprocessing task on the temp server

### PR DESCRIPTION
**Pull request by leseb**
With the help of https://github.com/instructlab/instructlab/pull/1050 we
have a working implementation of multiprocessing. We can then add again
the 'spawn' method when running the temporary server.

Signed-off-by: Sébastien Han <seb@redhat.com>

### CODE REVIEW

File: src/instructlab/server.py [hunk 1]
```diff
@@ -71,7 +71,7 @@ def ensure_server(
         return (None, None, None)
     except ClientException:
         tried_ports = set()
-        mpctx = multiprocessing.get_context(None)
+        mpctx = multiprocessing.get_context(DEFAULT_MULTIPROCESSING_START_METHOD)
```
Comment (russellb) [inline src/instructlab/server.py:74]: Are you sure this changes behavior?  When passing `None`, it should use the currently set default context. In other words, I don't think this change does anything.  Can you let me know if I'm wrong?

**Review by russellb — CHANGES_REQUESTED** [review]

File: src/instructlab/server.py (hunk 1 above)
Comment (leseb) [reply src/instructlab/server.py:74]: That's correct `None` will point to the default context. On Darwin, it's `spawn`, on Linux it's `fork`. We want to force `spawn` on all platforms.

@tiran please correct me if I'm wrong.

Comment (tiran) [reply src/instructlab/server.py:74]: Multiprocessing has two different types of defaults. There is the initial default setting for the context and there is the global default context. On Linux, the initial default is `fork` for Python <= 3.13. Python 3.14 will use spawn. We force-set `spawn` everywhere and override the value for `get_context(None)`:

```python
>>> import multiprocessing
>>> multiprocessing.get_context(None)
<multiprocessing.context.ForkContext object at 0x7f40a5b4bf50>
>>> multiprocessing.set_start_method("spawn", force=True)
>>> multiprocessing.get_context(None)
<multiprocessing.context.SpawnContext object at 0x7f40a5b4bfe0>
```

Comment (leseb) [reply src/instructlab/server.py:74]: Oh ok so this is the override:

https://github.com/instructlab/instructlab/blob/167ccd81f83dafbeb58d5f48fa2e87b2b7d45a6d/src/instructlab/lab.py#L31-L33

I'm closing this PR then. Thanks all!

**Pull request closed without merging**